src/lib/useReactDataTableState/useReactDataTableState.ts

```diff
@@ -0,0 +1,33 @@
+﻿import { useState } from "react";
+import { ColumnFiltersState, PaginationState, SortingState } from "@tanstack/react-table";
+import { useReactDataTableStateProps } from "./useReactDataTableStateProps";
+import { useReactDataTableStateResult } from "./useReactDataTableStateResult";
+
+/**
+ * A custom hook that will initialize all the state needed for the react data table
+ * @returns the state and the setters
+ */
+const useReactDataTableState = (props: useReactDataTableStateProps = {}): useReactDataTableStateResult => {
+  const { initialColumnFilters, initialSorting, initialPagination } = props;
+
+  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>(initialColumnFilters ?? []);
+  const [enteredColumnFilters, setEnteredColumnFilters] = useState<ColumnFiltersState>(initialColumnFilters ?? []);
+  const [sorting, setSorting] = useState<SortingState>(initialSorting ?? []);
+  const [pagination, setPagination] = useState<PaginationState>({
+    pageIndex: initialPagination?.pageIndex ?? 0,
+    pageSize: initialPagination?.pageSize ?? 10,
+  });
+
+  return {
+    sorting,
+    pagination,
+    columnFilters,
+    enteredColumnFilters,
```

Comment:
@neotob 

Is the magnifying glass button still being drawn? If yes than i think we should search for a better name for this.

Reply:
the searchbutton is still drawn yes, i was thinking about "activeFilter" too :thinking:

Reply:
as discussed: `afterSearchFilter` or `searchedFilter`

Reply:
renamed 💪